bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -398,6 +361,59 @@ pub mod pallet {
 				.reanchored(&T::BridgeHubLocation::get(), &T::UniversalLocation::get())
 				.map_err(|_| Error::<T>::LocationConversionFailed)
 		}
+
+		fn swap_fee_asset_and_burn(
+			origin: Location,
+			fee_asset: Asset,
+		) -> Result<u128, DispatchError> {
+			let ether_location = T::EthereumLocation::get();
+			let (fee_asset_location, fee_amount) = match fee_asset {
+				Asset { id: AssetId(ref loc), fun: Fungible(amount) } => (loc, amount),
+				_ => {
+					tracing::debug!(target: LOG_TARGET, ?fee_asset, "error matching fee asset");
+					return Err(Error::<T>::UnsupportedAsset.into())
+				},
+			};
+			if fee_amount == 0 {
+				return Ok(fee_amount)
+			}
```

Comment:
this could go back to `ensure!(fee_amount > 0)`

Reply:
I'd assume it's a valid case to register the asset with `fee_amount` set to zero, and later have someone else add a tip to the message, as you mentioned?